pkg/sentry/arch/arch_aarch64.go

```diff
@@ -97,8 +97,8 @@ type State struct {
 	// Our floating point state.
 	aarch64FPState `state:"wait"`
 
-	// TLS pointer
-	TPValue uint64
+	// Arch-specific register: tpidr_el0.
+	ArchReg uint64
```

Comment:
can we add this register into the Registers structure which is arch-specific?

Reply:
Hmmm...
I also wanted to do this at first.....
At first I made a modification according to this idea,  and it will destroy the unified logic of 'switchToApp' in kvm/ptrace platform.
Such as, in ptrace/subprocess.go:switchToApp().
The logic is like following:
regs := &ac.StateData().Regs
tls := uint64(ac.TLS())
fpState := ac.FloatingPointData()
...
t.setRegs(regs);
t.setFPRegs(fpState)
t.setTLS(&tls)
...
If I made a modification according to this idea, this logic code will become platform-dependent.
In Arm64, the logic is like following:
regs := &ac.StateData().Regs.commonReg
And in Amd64, the logic is same as before.
regs := &ac.StateData().Regs
The kvm platform also needs to make similar changes.

Reply:
In Linux kernel, the full context related registers of Arm64 are indeed stored in these 3 parts: PtraceRegs, tpidr_el0, fpstate.

Reply:
I am testing a new method for it, I will update it later

Reply:
What do you think of this new revision? @avagin